clippy_lints/src/methods.rs

```diff
@@ -1398,6 +1424,144 @@ fn lint_unnecessary_fold(cx: &LateContext<'_, '_>, expr: &hir::Expr, fold_args:
     };
 }
 
+mod unnecessary_filter_map {
```

Comment:
Do you think it's possible to move this to its own file? `methods.rs` is already quite large.